### DISCUSSION TITLE
Fritz!BOX 4020 Help

### DISCUSSION THREAD

**Discussion by cpcker** [xchwarze/wifi-pineapple-cloner#22] [Q&A]
Hi I will be laying soon hands on a fritz!box 4020 can i get some info and help beforehand on how to install since it has no USB ports 
and any other info that i will be needing thanks in advance

**Comment by xchwarze** [comment]
The project can be run on any hardware that meets the following requirements
https://github.com/xchwarze/wifi-pineapple-cloner/blob/master/build.md

For practicality's sake, I have only compiled a list of the most common ones available.

**Reply by cpcker** [reply]
the 4020 is supported by the project but since im new to flashing routers etc can I get some help on building the image I am a bit confused with all the the README files
Thanks in Advance!

**Reply by xchwarze** [reply]
use a virtual machine with ubuntu 20 or 22 and follow the steps of the automatic build. 
everyone who needs to do a custom build does it with that setup and those steps. i really don't know in which variant of the sdk is that target or what name it has so i can't give you the last line for your router.

```
chmod +x tools/*.sh
tools/dependencies-install.sh openwrt-deps-mips
sudo tools/dependencies-install.sh ubuntu-deps
tools/builder.sh mips universal imagebuilder/19.07.7-ar71xx-generic gl-ar750s
```
